server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -272,6 +296,64 @@ public void onFailure(Exception e) {
         }
     }
 
+    static Map<String, IndexTemplateV2.DataStreamTemplate> resolveAutoCreateDataStreams(Metadata metadata,
```

Comment:
That is trappy indeed, thanks for spotting this. I will update the PR, so that most of the auto create logic here is moved to the to be built auto create action.